dev/doc/changes.md

```diff
@@ -150,7 +150,33 @@ END
 
 #### VERNAC EXTEND
 
-Not handled yet.
+Steps to perform:
+- replace the brackets enclosing OCaml code in actions and rule classifiers with
+  braces
+- if not there yet, add a leading `|̀  to the first rule
+
+Handwritten classifiers declared through the `CLASSIFIED BY` statement are
+considered OCaml code, so they also need to be wrapped in braces.
+
+All extension macros are now behaving as if they were declared `FUNCTIONAL`,
+which means that they must return an interpretation state. This should
+currently be done by merely returning the `st` variable that is implicitly
+passed to the rule code.
+
+For instance, code of the form:
+```
+VERNAC COMMAND EXTEND my_command CLASSIFIED BY classifier
+  [ "foo" int(i) ] => [ classif' ] -> [ cmd1 i ]
+| [ "bar" ] -> [ cmd2 ]
+END
+```
+should be turned into
+```
+VERNAC COMMAND EXTEND my_command CLASSIFIED BY { classifier }
+  [ "foo" int(i) ] => { classif' } -> { let () = cmd1 i in st }
+| [ "bar" ] -> { let () = cmd2 in st  }
```

Comment:
can't you add this "context" in your preprocessor? I don't see why forcing people to return an unbound variable can be helpful (today and even tomorrow when st will be bound for real).

Reply:
Said otherwise, I'd have `VERNAC....` that is the one that expects the st to be returned and `LEGACY VERNAC ...` that is supposed to die and adds the `let () = ... in st` wrapper. So that when you remove LEGACY you get a sensible type error.

Reply:
I didn't really understand @ejgallego's plan on this feature. The current situation does not make sense at all, because the returned state is never taken into account by the vernac interpretation. Adding a LEGACY marker would ease the transition, but the default case will require fixing anyways as it is broken...

Reply:
In order to handle locality we had to change the extension type, so I opted for a quite general one instead of having to define many. That being said feel free to refine with attributes or something.

Reply:
In other words the current code is `LEGACY` and there is no infrastructure to remove the `LEGACY` (given that the result of functional code is discarded).

In any case, I prefer the code written between `{}` to make sense (in the old imperative model or in the new functional and only half baked mode). This PR makes the code just look silly, so please hide the silly context in the macro (with or without LEGACY, this is up to you).

Reply:
I wouldn't worry too much about this, but if you want, make the return type an option type, so the top level interpreter can know what to do (`None` would mean legacy)

Anyways, I feel that problems with the ongoing migration to pure state handling shouldn't delay this PR too much.